glusterd2/brick/validation.go

```diff
@@ -109,7 +111,9 @@ func validateXattrSupport(brickPath string) error {
 	return unix.Setxattr(brickPath, testXattrKey, []byte("payload"), 0)
 }
 
-func validateBrickInUse(brickPath string) error {
+// validateBrickWasUsed checks if the path was ever used a brick for a volume
+// by checking if the path has certain glusterfs specific xattrs
+func validateBrickWasUsed(brickPath string) error {
```

Comment:
question:
In which use case this function will be used if check for brick in use is mandatory?

Reply:
Use case: if you create a volume and then delete it, the xattr on the brick path isn't removed. When you try to create another volume from bricks of previously deleted volume, that check will be useful.

Reply:
if we delete volume. don't we need to delete brick path also?

Reply:
> if we delete volume. don't we need to delete brick path also?

No we don't. Deleting the volume doesn't delete the data present in it. It also doesn't remove xattr present on brick path to prevent users accidentally reusing it without warning

Reply:
if this is the behavior in gd1, then its okay to have this.